src/main/types/SIOP.types.ts

```diff
@@ -207,200 +239,204 @@ export interface AuthenticationResponseWithJWT {
   jwt: string;
   nonce: string;
   state: string;
+  idToken: IdToken;
   payload: AuthenticationResponsePayload;
   verifyOpts?: VerifyAuthenticationRequestOpts;
   responseOpts: AuthenticationResponseOpts;
 }
 
-export interface RequestRegistrationOpts extends RPRegistrationMetadataOpts {
-  registrationBy: RegistrationType;
-
-  // slint-disable-next-line @typescript-eslint/no-explicit-any
-  // [x: string]: any;
-}
-
-export interface DiscoveryMetadataOpts {
-  //TODO add the check: Mandatory if PassBy.Value
-  vpFormats?: Format;
+interface DiscoveryMetadataCommonOpts {
   //TODO add the check: Mandatory if PassBy.Value
-  authorizationEndpoint?: Schema.OPENID | string;
-  tokenEndpoint?: string;
-  userinfoEndpoint?: string;
+  authorizationEndpoint?: Schema | string;
+  // this is a confusion point. In the interop profile it mentions "https://self-issued.me/v2/openid-vc", but in the SIOPv2 it's mentioning "https://self-issued.me/v2"
+  // @Niels also created an issue here: https://github.com/decentralized-identity/jwt-vc-presentation-profile/issues/63 so we can keep an eye on this for clarification
   //TODO add the check: Mandatory if PassBy.Value
-  issuer?: ResponseIss;
-  jwksUri?: string;
-  registrationEndpoint?: string;
+  issuer?: ResponseIss | string;
   //TODO add the check: Mandatory if PassBy.Value
-  responseTypesSupported?: ResponseType[] | ResponseType;
-  responseModesSupported?: ResponseMode[] | ResponseMode;
-  grantTypesSupported?: GrantType[] | GrantType;
-  acrValuesSupported?: AuthenticationContextReferences[] | AuthenticationContextReferences;
-  scopesSupported?: Scope[] | Scope;
-  subjectTypesSupported?: SubjectType[] | SubjectType;
+  responseTypesSupported?: ResponseType[];
```

Comment:
According to the specs: 

response_types_supported

REQUIRED. A JSON array of strings representing supported response types. MUST be id_token.

Reply:
Yes but our implementation ensures max interop. Because in a lot of these cases a singular object is also being used next to arrays. 
Whenever you start removing stuff that is present in current implementation you should start asking questions first.

Reply:
Fixed